demos/encrypted-vm-image/README.md

```diff
@@ -0,0 +1,66 @@
+
+Firstly, a general workflow of deploying a confidential VM image is presented from the Guest Owner's perspective. Here, the *Guest Owner* is the client in a VM-based TEE environment that would like to use the confidential computing cloud.
+Then a step-by-step guide is given to show how the guest owner can prepare the encrypted image.
+
+## High-level Workflow
+
+The following diagram illustrates the high-level workflow and interdependencies between components.
+
+![](./workflow.jpeg)
+
+The workflow of this demo can be described as five steps.
+
+- Step 1: We use scripts to generate a protected guest image from a base image. Specifically, a new image that includes an EFI partition, a boot partition, and the most crucial one - a root partition is created. To safeguard the root partition (RootFS), it should be transformed into a JinDisk format using a pre-set disk encryption key. Additionally, necessary components (initramfs hooks, RA scripts, and other software dependencies) should be included to enable the guest VM startup.
+
+- Step 2: The guest owner uses QEMU/KVM to launch a secure VM through a TEE-supported Hypervisor (such as QEMU/KVM). The Hypervisor will help calculate the measurement of the guest VM's kernel and report to the guest owner, ensuring that an expected VM image is launched.
+
+- Step 3: The initramfs hooks request the attestation report when the kernel is booted. For example in AMD's SEV-SNP, a guest VM can make use of the SEV-SNP hardware to obtain an attestation report via the `sevguest.ko` and `ccp.ko` kernel modules ([CCP](https://lwn.net/Articles/735732/)). The `ccp.ko` module can optionally store the VCEK certificate for the platform along with the certificate chain necessary to validate the VCEK certificate. This guest kernel driver is also responsible for sending the `SNP_GUEST_REQUEST` message to the ASP firmware and presenting the reply back to the user space. Intel TDX also provides similar measurements (just like SEV's attestation report) for the guest owner to verify.
+
+- Step 4: The initramfs hooks send the attestation report to the guest owner or a key server. If the verification is passed, a trusted communication channel can be built. The cryptographic key to decrypt the root partition is retrieved by the guest VM.
+
+- Step 5: The initramfs hooks decrypt the guest VM's root filesystem with the key. Before unlocking, the root partition should be mounted automatically during the kernel boot. Then, the pre-installed initramfs hooks invoke the functionalities in the `jindisksetup` automatically to open the JinDisk-formatted root partition. 
+
+## Step-by-step Workflow
```

Comment:
I am a bit of confused as to why this step-by-step workflow (has three steps)seems to be even simpler than the high-level workflow (has five steps).